app/src/main/java/ai/elimu/vitabu/ui/storybook/StoryBookActivity.kt

```diff
@@ -47,6 +53,42 @@ class StoryBookActivity : AppCompatActivity() {
 
         val zoomOutPageTransformer = ZoomOutPageTransformer()
         binding.viewPager.setPageTransformer(true, zoomOutPageTransformer)
+        binding.viewPager.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
```

Comment:
```suggestion
```

@tuancoltech Instead of adding more code to the `onCreate()` method, would something like this work?:

```kotlin
@Override
protected void onDestroy() {
    super.onDestroy();
    
    if (viewPager != null) {
        int currentPosition = viewPager.getCurrentItem();
```

Reply:
@tuancoltech True, that will happen frequently the way the book navigation is currently implemented. And I've made a separate issue for preventing that:
* https://github.com/elimu-ai/vitabu/issues/272

Reply:
@tuancoltech Instead of adding two new variables and more code to the onCreate() method, would something like this work?:

```kotlin
@Override
protected void onDestroy() {
    super.onDestroy();
    
    if (viewPager != null) {
        int currentPosition = viewPager.getCurrentItem();
        .
        .
        .
```